frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -579,6 +604,39 @@ common_settings.document_metadata_location = {
         },
         genMetadataFolderMenuItem("doc"),
         genMetadataFolderMenuItem("dir"),
+        genMetadataFolderMenuItem("hash"),
+        { -- hash-based metadata count / TextViewer
+            text_func = function()
+                local hash_text = _("Show documents with hash-based metadata")
+                local no_hash_text = _("No documents with hash-based metadata")
+                if DocSettings.isHashLocationEnabled() then
+                    if G_reader_settings:readSetting("document_metadata_folder") ~= "hash" then
+                        return  "⚠ "..hash_text
+                    end
+                    return  hash_text
+                end
+                return no_hash_text
+            end,
+            keep_menu_open = true,
+            enabled_func = function()
+                return DocSettings.isHashLocationEnabled()
+            end,
+            callback = function()
+                local hash_file_infos = DocSettings.getHashDirSdrInfos()
+                local book_info_items = {}
+                for i, file_info in ipairs(hash_file_infos) do
+                    table.insert(book_info_items, table.concat({"\n", i, ". ", file_info}))
+                end
+                local book_info_str = table.concat(book_info_items)
+                UIManager:show(require("ui/widget/textviewer"):new{
+                    title = T(N_("1 document with hash-based metadata", "%1 documents with hash-based metadata",
+                        #hash_file_infos), #hash_file_infos),
+                    title_multilines = true,
+                    justified = false,
+                    text = hash_metadata_file_list_header..book_info_str,
```

Comment:
I'm trying not to nit but earlier you unnecessarily added to the diff by adding spaces around `..` and here there are none! :laughing:

Reply:
Yes someone wanted them so I went through a bunch of places but probably missed some. I'll see if there's any others lol

Reply:
I fixed the spacing of various other `..` that were previously also wrong.

Reply:
I often find no spaces more natural to read really (strongly context dependent); it just really stood out to me because several of the first lines I saw in the diff were nothing but adding some spaces.  ;-)

Reply:
I personally like no spaces more for concatenation but don't know what general Lua conventions are. Fine to do whatever the maintainers prefer though :)

Reply:
I would expect with spaces to be more common but I haven't verified that. I also know plenty that don't use any.

I think we (or maybe just me) take the view that `little.."thing"` and `"little".."thing"` are contextually very clear while `"long long long long" .. "long long long"` might get lost in the sea without spaces.